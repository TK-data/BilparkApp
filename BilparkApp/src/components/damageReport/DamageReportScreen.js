import React from 'react';
import { StyleSheet, View, Image, Dimensions } from 'react-native';
import { Text } from 'native-base';

import DamageReportForm from './DamageReportForm';

const image = require('../../images/menuIcons/car_repair.png');

const window = Dimensions.get('window');

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#002776',
    // paddingTop: '5%',
    // paddingBottom: '5%',
  },
  topContainer: {
    display: 'flex',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'red',
  },
  form: {
    display: 'flex',
    flex: 2,
    alignItems: 'center',
  },
  text: {
    color: 'white',
    fontSize: 22,
    fontWeight: 'bold',
  },
  titleContainer: {
    display: 'flex',
    alignItems: 'center',
    backgroundColor: 'white',
    width: window.width,
  },
  title: {
    color: 'rgb(000, 039, 118)',
    fontSize: 20,
    marginBottom: 5,
    marginTop: -15,
  },
  image: {
    width: window.width * 0.15,
    height: window.height * 0.15,
    zIndex: 5,
    margin: 0,
    marginTop: -10,
  },
});

const DamageReportScreen = () => (
  <View style={styles.container}>
    <View style={styles.titleContainer}>
      <Image resizeMode="contain" style={styles.image} source={image} />
      <Text style={styles.title}>Skaderegistrering</Text>
    </View>
    <View style={styles.form}>
      <DamageReportForm />
    </View>
  </View>
);

DamageReportScreen.navigationOptions = {
  title: 'Registrer skader p√• din bil',
};

export default DamageReportScreen;
