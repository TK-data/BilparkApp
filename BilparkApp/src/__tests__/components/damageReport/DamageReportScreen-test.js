import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import DamageReportScreen from '../../../components/damageReport/DamageReportScreen';

const initialState = {
};

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

describe('Testing DamageReportScreen', () => {
  it('Renders as expected', () => {
    const comp = shallow(
      <DamageReportScreen />,
      { context: { store: mockStore(initialState) } },
    );
    expect(comp.dive()).toMatchSnapshot();
  });
});
