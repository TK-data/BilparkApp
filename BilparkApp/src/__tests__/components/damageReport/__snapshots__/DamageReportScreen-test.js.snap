// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DamageReportScreen Renders as expected 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#002776",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "display": "flex",
          "width": 750,
        }
      }
    >
      <Image
        resizeMode="contain"
        source={
          Object {
            "process": [Function],
          }
        }
        style={
          Object {
            "height": 200.1,
            "margin": 0,
            "marginTop": -10,
            "width": 112.5,
            "zIndex": 5,
          }
        }
      />
      <Styled(Text)
        style={
          Object {
            "color": "rgb(000, 039, 118)",
            "fontSize": 20,
            "marginBottom": 5,
            "marginTop": -15,
          }
        }
        virtual={undefined}
      >
        Skaderegistrering
      </Styled(Text)>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "flex": 2,
        }
      }
    >
      <Connect(DamageReportForm) />
    </View>
  </View>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "display": "flex",
              "width": 750,
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "process": [Function],
              }
            }
            style={
              Object {
                "height": 200.1,
                "margin": 0,
                "marginTop": -10,
                "width": 112.5,
                "zIndex": 5,
              }
            }
          />
          <Styled(Text)
            style={
              Object {
                "color": "rgb(000, 039, 118)",
                "fontSize": 20,
                "marginBottom": 5,
                "marginTop": -15,
              }
            }
            virtual={undefined}
          >
            Skaderegistrering
          </Styled(Text)>
        </View>,
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 2,
            }
          }
        >
          <Connect(DamageReportForm) />
        </View>,
      ],
      "style": Object {
        "alignItems": "center",
        "backgroundColor": "#002776",
        "flex": 1,
        "justifyContent": "center",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 200.1,
                  "margin": 0,
                  "marginTop": -10,
                  "width": 112.5,
                  "zIndex": 5,
                }
              }
            />,
            <Styled(Text)
              style={
                Object {
                  "color": "rgb(000, 039, 118)",
                  "fontSize": 20,
                  "marginBottom": 5,
                  "marginTop": -15,
                }
              }
              virtual={undefined}
            >
              Skaderegistrering
            </Styled(Text)>,
          ],
          "style": Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "display": "flex",
            "width": 750,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "resizeMode": "contain",
              "source": Object {
                "process": [Function],
              },
              "style": Object {
                "height": 200.1,
                "margin": 0,
                "marginTop": -10,
                "width": 112.5,
                "zIndex": 5,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Skaderegistrering",
              "style": Object {
                "color": "rgb(000, 039, 118)",
                "fontSize": 20,
                "marginBottom": 5,
                "marginTop": -15,
              },
              "virtual": undefined,
            },
            "ref": null,
            "rendered": "Skaderegistrering",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(DamageReportForm) />,
          "style": Object {
            "alignItems": "center",
            "display": "flex",
            "flex": 2,
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "View",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "white",
                "display": "flex",
                "width": 750,
              }
            }
          >
            <Image
              resizeMode="contain"
              source={
                Object {
                  "process": [Function],
                }
              }
              style={
                Object {
                  "height": 200.1,
                  "margin": 0,
                  "marginTop": -10,
                  "width": 112.5,
                  "zIndex": 5,
                }
              }
            />
            <Styled(Text)
              style={
                Object {
                  "color": "rgb(000, 039, 118)",
                  "fontSize": 20,
                  "marginBottom": 5,
                  "marginTop": -15,
                }
              }
              virtual={undefined}
            >
              Skaderegistrering
            </Styled(Text)>
          </View>,
          <View
            style={
              Object {
                "alignItems": "center",
                "display": "flex",
                "flex": 2,
              }
            }
          >
            <Connect(DamageReportForm) />
          </View>,
        ],
        "style": Object {
          "alignItems": "center",
          "backgroundColor": "#002776",
          "flex": 1,
          "justifyContent": "center",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Image
                resizeMode="contain"
                source={
                  Object {
                    "process": [Function],
                  }
                }
                style={
                  Object {
                    "height": 200.1,
                    "margin": 0,
                    "marginTop": -10,
                    "width": 112.5,
                    "zIndex": 5,
                  }
                }
              />,
              <Styled(Text)
                style={
                  Object {
                    "color": "rgb(000, 039, 118)",
                    "fontSize": 20,
                    "marginBottom": 5,
                    "marginTop": -15,
                  }
                }
                virtual={undefined}
              >
                Skaderegistrering
              </Styled(Text)>,
            ],
            "style": Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "display": "flex",
              "width": 750,
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "resizeMode": "contain",
                "source": Object {
                  "process": [Function],
                },
                "style": Object {
                  "height": 200.1,
                  "margin": 0,
                  "marginTop": -10,
                  "width": 112.5,
                  "zIndex": 5,
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Skaderegistrering",
                "style": Object {
                  "color": "rgb(000, 039, 118)",
                  "fontSize": 20,
                  "marginBottom": 5,
                  "marginTop": -15,
                },
                "virtual": undefined,
              },
              "ref": null,
              "rendered": "Skaderegistrering",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(DamageReportForm) />,
            "style": Object {
              "alignItems": "center",
              "display": "flex",
              "flex": 2,
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "View",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
